src/Ensconce.Cli/ApplicationInteraction.cs

```diff
@@ -7,95 +7,131 @@
 using System.Management;
 using System.ServiceProcess;
+using System.Text.RegularExpressions;
 using System.Threading;
 
 namespace Ensconce.Cli
 {
     internal static class ApplicationInteraction
     {
+        private static readonly Regex HandleRegEx = new Regex(@"^(?<exe>.*?)\s*pid:\s*(?<pid>[0-9]+)\s*type:\s*(?<type>\w*)\s*?(?<hash>[A-Z0-9]+)\s*:\s*(?<path>.*)$");
+
         internal static void StopAndDeleteServicesInDirectory(string directory)
         {
-            foreach (var serviceDetails in GetServicesInstalledInDirectory(directory))
+            Logging.Log("Finding services to stop and delete in {0}", directory);
+            var services = GetServicesInstalledInDirectory(directory).ToList();
+            if (services.Any())
             {
-                var service = ServiceController.GetServices().First(svc => svc.DisplayName == serviceDetails.DisplayName);
-                if (service.Status != ServiceControllerStatus.Stopped)
+                Logging.Log("{0} services found in {1}", services.Count, directory);
+                foreach (var serviceDetails in services)
                 {
-                    Logging.Log("Stopping service {0}", serviceDetails.DisplayName);
-                    service.Stop();
-                }
+                    var service = ServiceController.GetServices().First(svc => svc.DisplayName == serviceDetails.DisplayName);
+                    if (service.Status != ServiceControllerStatus.Stopped)
+                    {
+                        Logging.Log("Stopping service {0}", serviceDetails.DisplayName);
+                        service.Stop();
+                    }
 
-                VerifyServiceStopped(serviceDetails.DisplayName, 30);
+                    VerifyServiceStopped(serviceDetails.DisplayName, 30);
 
-                Logging.Log("Uninstalling service {0}", serviceDetails.DisplayName);
-                Process.Start("sc", string.Format("delete \"{0}\"", serviceDetails.ServiceName)).WaitForExit();
+                    Logging.Log("Uninstalling service {0}", serviceDetails.DisplayName);
+                    Process.Start("sc", $"delete \"{serviceDetails.ServiceName}\"")?.WaitForExit();
 
-                VerifyServiceUninstall(serviceDetails.DisplayName, 30);
+                    VerifyServiceUninstall(serviceDetails.DisplayName, 30);
+                }
+            }
+            else
+            {
+                Logging.Log("No services found in {0}", directory);
             }
         }
 
         internal static void StopProcessesInDirectory(string directory)
         {
-            if (!directory.EndsWith(@"\"))
+            Logging.Log("Stopping processes in directory: {0}", directory);
+
+            var processes = GetProcessesRunningInDirectory(directory);
+            if (!directory.EndsWith(Path.DirectorySeparatorChar.ToString()))
             {
-                directory = directory + @"\";
+                directory += Path.DirectorySeparatorChar;
```

Comment:
No point manipulating this variable here - I think line 53 is supposed to be underneath this if-block

Reply:
Correct, i refactored it into the function that line53 calls.  but left it here to.
Removed